zokrates_core/src/flatten/mod.rs

```diff
@@ -2083,6 +2120,24 @@ impl<'ast, T: Field> Flattener<'ast, T> {
         }
     }
 
+    /// Unwraps a boolean & (and) expression chain to a flat vector of boolean expressions
+    ///
+    /// # Arguments
+    /// * `e` - `BooleanExpression` to be flattened
+    fn unwrap_and_chain(
```

Comment:
I think this should be in the `impl` of `BooleanExpression` as it only depends on the expression.

Also this looks a lot like an iterator and it's used as such, so maybe it's better to skip allocating into this vector and iterate straight out of the expression.

See an idea of what an iterator could look like [here](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018). The advantage is that its internal vector only grows when we see an `And`, as opposed to this one which allocates `vec![e]` for all elements.